lib/install/config/loaders/core/assets.js

```diff
@@ -1,11 +1,21 @@
-const { env, publicPath } = require('../configuration.js')
+const Url = require('url-parse')
+const { env, paths, publicPath } = require('../configuration.js')
+
+// Compute path for internal pack assets if ASSET_HOST is supplied
+const computeAssetPath = (host) => {
+  if (!host) { return `/${paths.entry}/` }
+  const { slashes, href } = new Url(host)
+  if (slashes) { return `${href}/${paths.entry}/` }
+  if (href) { return `//${href}/${paths.entry}/` }
+  return `/${paths.entry}/`
```

Comment:
Yepp, me don't like it either but seems we can't fully use `switch` as we are dependent on multiple params. I will break into two functions.